### PR TITLE
EX61/Product page

### PR DESCRIPTION
**Pull request by fbozhkov**
- Wrapped ProductCard in MUI Card component with product image being the link to the given product details page 
- Responsive image slider using pure-react-carousel
- Conditional rendering of section "sizes" depending on product type
- Saved scroll position on navigating back
- Fetch product info by product ID

### CODE REVIEW

**Review by petya0111 — APPROVED** [review]
Good work!

File: client/src/components/product-page/ProductPage.js [hunk 1]
```diff
@@ -0,0 +1,142 @@
...
+        .then ((response) => {
+            const categoryArray = response.data;
+            setCategory(categoryArray);
+            
+        })
+        .catch(error => console.error(`Error: ${error}`));
+    }
+
+    function hasSizes(productCategory) {
+        console.log('productCategory = ', productCategory );
```
Comment (IvayloSlavchev) [inline client/src/components/product-page/ProductPage.js]: Remove console.log

File: client/src/components/product-page/ProductPage.js [hunk 2]
```diff
@@ -0,0 +1,142 @@
...
+            return true;
+        }
+        else {
+            return false;
+        }
+    }
+    
+    const productCategory = category[product.type_id-1]?.name;
+    console.log('product', productCategory);
+    console.log('func', hasSizes(productCategory));
```
Comment (IvayloSlavchev) [inline client/src/components/product-page/ProductPage.js]: Same here

File: client/src/components/product-page/product-page.scss [hunk 3]
```diff
@@ -0,0 +1,109 @@
+@import'../../shared/styles/colors';
+
+.product-page {
+    max-width: 1536px; // equal to main-page layout width
+    min-height: 1000vh;
```
Comment (IvayloSlavchev) [inline client/src/components/product-page/product-page.scss]: We have to fix the width and height, bacause there is a lot of empty space.

File: client/src/components/product-page/ProductPage.js [hunk 4]
```diff
@@ -0,0 +1,195 @@
...
+        .then ((response) => {
+            const categoryArray = response.data;
+            setCategory(categoryArray);
+            
+        })
+        .catch(error => console.error(`Error: ${error}`));
+    }
+
+    function hasSizes(productCategory) {
+        /* console.log('productCategory = ', productCategory ); */
```
Comment (JordanRad) [inline client/src/components/product-page/ProductPage.js]: Please, remove the commented code

File: client/src/components/product-page/ProductPage.js [hunk 5]
```diff
@@ -0,0 +1,195 @@
...
+            return true;
+        }
+        else {
+            return false;
+        }
+    }
+    
+    const productCategory = category[product.type_id-1]?.name;
+    /* console.log('product', productCategory);
+    console.log('func', hasSizes(productCategory)); */
```
Comment (JordanRad) [inline client/src/components/product-page/ProductPage.js]: Same here

File: client/src/components/product-page/ProductPage.js [hunk 6]
```diff
@@ -0,0 +1,195 @@
...
+                            {hasSizes(productCategory) &&
+                                <Box className='sizes-div'>
+                                <Box className='sizes-text-div'>
+                                    <Typography className='select-size-text'>
+                                        {t('ProductPage.selectSize')}
+                                    </Typography>
+                                    {/* add size guide later (clickable) */}
+                                    {/* <Typography className='size-guide'>
+                                        Size Guide
+                                    </Typography> */}
```
Comment (JordanRad) [inline client/src/components/product-page/ProductPage.js]: Same here

**Review by JordanRad — APPROVED** [review]

File: client/src/components/product-page/product-page.scss (hunk 3 above)
Comment (fbozhkov) [reply client/src/components/product-page/product-page.scss]: For the height I agree. I am just waiting for the footer component and then I can style it to look good. As for the width I am following the style from tha main page layout. What do you suggest exactly?

File: client/src/components/product-page/ProductPage.js (hunk 1 above)
Comment (fbozhkov) [reply client/src/components/product-page/ProductPage.js]: removed

File: client/src/components/product-page/ProductPage.js (hunk 2 above)
Comment (fbozhkov) [reply client/src/components/product-page/ProductPage.js]: done

File: client/src/components/product-page/ProductPage.js (hunk 4 above)
Comment (fbozhkov) [reply client/src/components/product-page/ProductPage.js]: done

File: client/src/components/product-page/ProductPage.js (hunk 5 above)
Comment (fbozhkov) [reply client/src/components/product-page/ProductPage.js]: done

File: client/src/components/product-page/ProductPage.js (hunk 6 above)
Comment (fbozhkov) [reply client/src/components/product-page/ProductPage.js]: done

**Review by IvayloSlavchev — APPROVED** [review]
LGTM!

**Review by Arabadzhiew — APPROVED** [review]

**Pull request merged**